README.md

```diff
@@ -91,7 +91,7 @@ like `X-Forwarded-Proto: https` and
 You can use that to make HTTPS (`https://example.com/other-page`)
 redirections, or just use relative (`../other-page`) or protocol-agnostic
 (`//example.com/other-page`) redirections and it will always work
-anywhere (this is a good practice, BTW).
+anywhere (this should generally be avoided in any https only environment, so everywhere, it's 2019).
```

Comment:
This patch is wrong:

- There are sites that do not need to have (or, at least, enforce) HTTPS. Yes, it's better everywhere anyways, but i.e. you could be serving a site behind a VPN which would handle the encryption by itself, or you could be serving a blog generated by a SSG, where encryption wouldn't add really a lot.
- Almost every site in 2019 serves in both HTTP and HTTPS. The HTTP version of the site usually just returns a 301 redirection to the HTTPS one. You can test it clicking in http://amazon.com, http://google.com or http://github.com. But yes, there must be a server behind, listening in HTTP in port 80, to return that redirection.
- In an HTTPS-only site, using a protocol-agnostic link would render exactly the same results as using a protocol-specific link. Always.
- OTOH, if you start your site (i.e. your SSG blog) in HTTP and want to update later to HTTPS, using a protocol-agnostic address will make it work fine always.
- Saying things like "it's 2019" makes the README less maintainable. I would need to update it each year. Also it would imply that in 2018 the situation was different, which isn't the case. Also it seems to ridicule people who isn't using HTTPS this year, while I prefer a tone that everybody is comfortable to read.

Reply:
it's supposed to ridicule people who don't use https, there's no reason not to

https://www.paulirish.com/2010/the-protocol-relative-url/
"Now that SSL is encouraged for everyone and doesn’t have performance concerns, this technique is now an anti-pattern. If the asset you need is available on SSL, then always use the https:// asset."

Reply:
> it's supposed to ridicule people [...] there's no reason not to

![](https://media.giphy.com/media/l3q30Om7tCB8yHT0s/giphy.gif)